Sight_++_Solution/Intel_Sight_PP/src/Intel Sight ++.cpp

```diff
@@ -109,27 +112,47 @@ int main(int argc, char** argv)
 
 
 
-	pipe.start(cfg);
+	auto config = pipe.start(cfg);
 
 	std::cout << "Got pipeline\n";
 
 	MLController ml_controller;
 	// TODO Read command line parameters for which models to use?
+	
 	MLImplDepth ml_depth;
 	MLImplRGB ml_rgb;
+
+	
 	// TODO Add correct paths for testing
 	// TODO Add command line parameter for files to use?
-	CaffeModelImpl caffe("link to prototxt file", "link to caffemodel file", "link to class name text file");
-
+	
+	auto profile = config.get_stream(RS2_STREAM_COLOR).as<rs2::video_stream_profile>();
+	CaffeModelImpl caffe("./models/no_bn.prototxt", "./models/no_bn.caffemodel", "./models/no_bn-classnames.txt", profile);
+	
 	// Add more ML implementations here as needed
-	ml_controller.add_depth_model(ml_depth);
-	ml_controller.add_rgb_model(ml_rgb);
-	ml_controller.add_rgb_model(caffe);
+	ml_controller.add_model(ml_depth);
+	ml_controller.add_model(ml_rgb);
+	ml_controller.add_model(caffe);
 
 	std::cout << "Created MLController and added ML models\n";
 
-	Prioritiser prioritiser;
+	std::string name_prio_depth = "depth_prio";
+	depth_priority* prio_depth = new depth_priority(&name_prio_depth);
```

Comment:
Only thing I've noticed (and one, in line 148, and in priority_module.hpp), generally using new to create objects is not recommended in modern ++, see [this](https://stackoverflow.com/questions/6337294/creating-an-object-with-or-without-new) or [this](https://stackoverflow.com/questions/6500313/why-should-c-programmers-minimize-use-of-new).

Reply:
The cure of a java based undergrad